### PR TITLE
Restore source/targetCompatibility declarations for intellij to set version correctly and remo…

### PR DESCRIPTION
**Pull request by anuraaga**
…ve animalsniffer for java 8 since we set the `release` option now.

Thanks @tylerbenson for the callout.

### CODE REVIEW

File: sdk_extensions/jaeger_remote_sampler/build.gradle [hunk 1]
```diff
@@ -25,7 +25,6 @@ dependencies {
 
     testRuntime "io.grpc:grpc-netty-shaded:${grpcVersion}"
 
-    signature "org.codehaus.mojo.signature:java18:1.0@signature"
     signature "net.sf.androidscents.signature:android-api-level-24:7.0_r2@signature"
```
Comment (tylerbenson) [inline sdk_extensions/jaeger_remote_sampler/build.gradle:29]: Is it worth trying to put this string in a common place?

**Review by tylerbenson — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -169,6 +169,9 @@ configure(opentelemetryProjects) {
         mavenLocal()
     }
 
+    sourceCompatibility = 1.8
```
Comment (Oberon00) [inline build.gradle:172]: Could we use the new-style declarations as documented there? https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension ?

```java
java {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
}
```

Comment (anuraaga) [reply build.gradle]: Sure

File: sdk_extensions/jaeger_remote_sampler/build.gradle (hunk 1 above)
Comment (anuraaga) [reply sdk_extensions/jaeger_remote_sampler/build.gradle:29]: I think we have issue #1808 for that, let me look at it separately from this PR.

**Review by Oberon00 — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**